src/python/WMCore/Services/WMStats/WMStatsReader.py

```diff
@@ -332,6 +341,47 @@ def getRequestByStatus(self, statusList, jobInfoFlag=False, limit=None, skip=Non
 
         return results
 
+    def getRequestByStatusAndStartTime(self, statusList, jobInfoFlag=False, detail=False, startTime=0):
+        """
+        This is just a wrapper method to call getRequestByStatusAndStartTime from
+        request DB. It queries for requests that are in a specific status since startTime.
+        For this one to work we need reqDBURL to be set and reqDB object created.
```

Comment:
Hmm.. I guess T0 isn't defining it? If it needs to, then I'd suggest to make that parameter in the constructor method to be mandatory, such that we always have such object properly created.

Reply:
My opinion is the same, I just did not want to change an already established behavior. I did check and for us it is always the case - we do use those two in parallel and we never create the `wmstats` object without the `reqDB` one. So I agree with you. This would be the best. in this case.

Reply:
And as of:
> Hmm.. I guess T0 isn't defining it?

They do. And the `centralRequestDB` object is created  at the very top at CleanCouchPoller:
https://github.com/dmwm/WMCore/blob/master/src/python/WMComponent/TaskArchiver/CleanCouchPoller.py#L108-L122

But actually if we would want that to work properly, then it should be also passed to the `wmstatsCouchDB` object creation, few lines  above:
https://github.com/dmwm/WMCore/blob/a6bdba3dc88e88da044bf19882d629b50be21c55/src/python/WMComponent/TaskArchiver/CleanCouchPoller.py#L107